marshmallow/fields.py

```diff
@@ -395,7 +400,7 @@ def schema(self):
         .. versionchanged:: 1.0.0
             Renamed from `serializer` to `schema`
         """
-        if not self.__schema:
+        if not (self.__schema and self.withcache):
```

Comment:
This pull-request completely breaks nested schemas, because when self.withcache == False,self.__schema will not be initialized at all. Am I getting something wrong? :)

Reply:
The logic is fine.

|   | withcache | not withcache |
|---|---|---|
| **no __schema** | True 👈 | True 👈 |
| **__schema** | False | True |

Negating grouped expressions can be hard to read though. It is better to expand the expression so that when read aloud it makes sense.

"Make a schema if there is no schema or the cache is disabled" -> `if not self.__schema or not self.withcache`

---

That said, I'm still not a fan of making this a configurable option.